### PR TITLE
Improve doco around azure with bicep scripts.

### PR DESCRIPTION
**Pull request by LukeButters**
# Background

Improve doco around bicep script.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: testing/compatibility/azure/deploy.ps1 [hunk 1]
```diff
@@ -10,76 +10,85 @@ $env:adminPassword = $adminPassword
...
 #---------------------------------
 $env:tentacleUri = "https://octopus-downloads-staging.s3.amazonaws.com/octopus/Octopus.Tentacle.$env:tentacleVersion-net6.0-win-x64.msi"
+$netFramworkVersion = 'net6'
 
 # Windows 2012
 $env:virtualMachineName = 'Net6Tentacle-Window2012'
 $env:virtualMachineSku = '2012-datacenter-gensecond'
 $env:os = 'windows2012'
-az deployment group create --name Tentacle --resource-group Tentacle --template-file vm.bicep --parameters vm.bicepparam
+az deployment group create --name $group --resource-group $resourcegroup --template-file vm.bicep --parameters vm.bicepparam
```
Comment (LukeButters) [inline testing/compatibility/azure/deploy.ps1:32]: This makes it easy to change resource group

File: testing/compatibility/azure/deploy.ps1 [hunk 2]
```diff
@@ -10,76 +10,85 @@ $env:adminPassword = $adminPassword
...
 
 # Windows 2022
 $env:virtualMachineName = 'Net48Tentacle-Window2022'
 $env:virtualMachineSku = '2022-datacenter-azure-edition'
 $env:os = 'windows2022-48'
-az deployment group create --name Tentacle --resource-group Tentacle --template-file vm.bicep --parameters vm.bicepparam
\ No newline at end of file
+az deployment group create --name $group --resource-group $resourcegroup --template-file vm.bicep --parameters vm.bicepparam
+
+echo "Net48 tentacles wont connect, restart all net48 VMs then re-run the script in c:\\TentacleInStallRun.ps1 OR after restart modify the install-tentacle.ps1 script (add an extra Write-Host) so bicep see's it as a differnt script and so will re-run the script on the VMs again."
```
Comment (LukeButters) [inline testing/compatibility/azure/deploy.ps1:94]: Net48 wont work but lets tell the user what they can do

File: testing/compatibility/azure/install-tentacle.ps1 [hunk 3]
```diff
@@ -21,19 +27,20 @@ if ($installedVersion -lt $net48BuildNumber) {
     $process = start-process "C:\Windows\Temp\ndp48-x86-x64-allos-enu.exe" -argumentlist @("/q", "/norestart", "/log", "C:\Windows\Temp\ndp48-x86-x64-allos-enu.log") -wait -PassThru
     $process.WaitForExit()
 	Write-Host "Installed .NET 4.8"
+    # For net48 to work the machine must be rebooted and this scripted edited again.
```
Comment (LukeButters) [inline testing/compatibility/azure/install-tentacle.ps1]: More hints.

File: testing/compatibility/azure/vm.bicep [hunk 4]
```diff
@@ -125,7 +126,7 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2022-03-01' = {
   location: location
   properties: {
     hardwareProfile: {
-      vmSize: 'Standard_B2als_v2'
+      vmSize: 'Standard_DS1_v2'
```
Comment (LukeButters) [inline testing/compatibility/azure/vm.bicep:129]: It seems we can actually get the number of CPUs to be increased in azure subscription.

File: testing/compatibility/azure/deploy.ps1 [hunk 5]
```diff
@@ -10,76 +10,85 @@ $env:adminPassword = $adminPassword
 
 $env:octopusServerRole = 'OsTestingListening'
 $env:octopusServerEnvironment = 'OsTesting'
-$env:tentacleVersion = '7.0.201'
+$env:tentacleVersion = '8.1.32'
 
+## Takes this on to the end of the tentacle's name as seen in octopus
+$env:tentacleNamePostfix = "$env:tentacleVersion"
```
Comment (LukeButters) [inline testing/compatibility/azure/deploy.ps1:16]: This means its really easy to see what the tentacle is from the octopus side.

Comment (sburmanoctopus) [reply testing/compatibility/azure/deploy.ps1:16]: Nit: I don't think `postfix` is a real word. Suffix?

File: testing/compatibility/azure/deploy.ps1 [hunk 6]
```diff
@@ -10,76 +10,85 @@ $env:adminPassword = $adminPassword
...
 
 # Windows 2022
 $env:virtualMachineName = 'Net48Tentacle-Window2022'
 $env:virtualMachineSku = '2022-datacenter-azure-edition'
 $env:os = 'windows2022-48'
-az deployment group create --name Tentacle --resource-group Tentacle --template-file vm.bicep --parameters vm.bicepparam
\ No newline at end of file
+az deployment group create --name $group --resource-group $resourcegroup --template-file vm.bicep --parameters vm.bicepparam
+
+echo "Net48 tentacles wont connect, restart all net48 VMs then re-run the script in c:\\TentacleInstallRun.ps1 OR after restart modify the install-tentacle.ps1 script (add an extra Write-Host) so bicep see's it as a differnt script and so will re-run the script on the VMs again."
```
Comment (sburmanoctopus) [inline testing/compatibility/azure/deploy.ps1:94]: Nit: wont => won't => will not

File: testing/compatibility/azure/install-tentacle.ps1 [hunk 7]
```diff
@@ -4,7 +4,13 @@ $octopusServerApiKey,
 $octopusServerRole,
 $octopusServerEnvironment,
 $os,
-$tentacleUri)
+$tentacleUri,
+$tentacleNamePostfix)
+
+
+myargs = "$octopusServerThumbprint $octopusServerUrl $octopusServerApiKey $octopusServerRole $octopusServerEnvironment $os $tentacleUri $tentacleNamePostfix"
```
Comment (sburmanoctopus) [inline testing/compatibility/azure/install-tentacle.ps1:11]: Nit: Maybe add a comment as to what this is doing and why it is here.

I helped you with some of the powershell, and I initially forgot... 😁

File: testing/compatibility/azure/readme.md [hunk 8]
```diff
@@ -40,11 +40,12 @@ Ensure the variables in `deploy.ps1` are correct. Some variables of note:
 
 ### Deploy
 Now, install the VMs and install Tentacle:
-> ./deploy.ps1 -octopusServerThumbprint {OctopusServerThumbprint} -octopusServerUrl {OctopusServerUri} -octopusServerApiKey {OctopusServerApiKey} -adminPassword qwerty123!@#
+> ./deploy.ps1 -octopusServerThumbprint {OctopusServerThumbprint} -octopusServerUrl {OctopusServerUri} -octopusServerApiKey {OctopusServerApiKey} -adminPassword {superSecretPasswordToTheVms.2}
```
Comment (sburmanoctopus) [inline testing/compatibility/azure/readme.md:43]: Why the `.2` at the end of `superSecretPasswordToTheVms.2`?

**Review by sburmanoctopus — APPROVED** [review]
LGTM. But I'm not the Bicep guru. Probably worth getting @nathanwoctopusdeploy on this one 😁

File: testing/compatibility/azure/readme.md (hunk 8 above)
Comment (LukeButters) [reply testing/compatibility/azure/readme.md:43]: Password requirements of needing a symbol and number.

File: testing/compatibility/azure/deploy.ps1 (hunk 6 above)
Comment (LukeButters) [reply testing/compatibility/azure/deploy.ps1]: done

File: testing/compatibility/azure/install-tentacle.ps1 (hunk 7 above)
Comment (LukeButters) [reply testing/compatibility/azure/install-tentacle.ps1:11]: Done

**Pull request merged**